docs/release.md

```diff
@@ -1,35 +1,47 @@
-## Releasing
+## Releases
 
-This document outlines how to cut a new release.
+### Release cadence
 
-### Checklist
+**reth does not currently have a regular release cadence while it is still experimental software**.
+
+### For maintainers
+
+This section outlines how to cut a new release.
 
 It is assumed that the commit that is being considered for release has been marked as stable, i.e. that there is an expectation of no major bugs.
 
-- [ ] Ensure *all* tests and lints pass
-- [ ] Changelog
-  - [ ] Ensure the changelog is up to date by going over https://github.com/paradigmxyz/reth/labels/M-changelog
-  - [ ] Move the "Unreleased" section in [CHANGELOG.md](./CHANGELOG.md) into a header with the new version, and add a link at the bottom of the changelog to compare the new version to HEAD.
+#### Release PR
+
+- [ ] Create a new branch (e.g. `release/vx.y.z`) and open a pull request for it
+- [ ] Ensure *all* tests and lints pass for the chosen commit
 - [ ] Version bump
-  - [ ] Update the version in `Cargo.toml`
+  - [ ] Update the version in *all* `Cargo.toml`'s
 - [ ] Commit the changes
-  - The message format should be `release: x.y.z`, substituting `x.y.z` for the semver.
-- [ ] Tag the new commit with `x.y.z`
-- [ ] Push the changes, including the tag
+  - The message format should be `release: vx.y.z`, substituting `x.y.z` for the semver.
+- [ ] The PR should be reviewed to see if anything was missed
+- [ ] Once reviewed, merge the PR
 
-After pushing the tag, CI will start building the release, bundling it, and publishing it to package managers and crates.io. (TBD)
+#### Releasing
+
+- [ ] Tag the new commit on main with `vx.y.z` (`git tag vx.y.z SHA`)
+- [ ] Push the tag (`git push origin vx.y.z`)[^1]
+- [ ] Run the release commit on testing infrastructure for 1-3 days to check for inconsistencies and bugs
+  - This testing infrastructure is going to sync and keep up with a live testnet, and includes monitoring of bandwidth, CPU, disk space etc.
 
 > **Note**
```

Comment:
We should also add releases for the crates, for library users. I've been using this script in Ethers-rs: https://github.com/gakonst/ethers-rs/blob/master/bin/release. We could also use release-plz: https://github.com/MarcoIeni/release-plz which is recent and seems very promising.

Reply:
Let's do that in a follow up